app/views/catalog/_password_modal.html.erb

```diff
@@ -5,17 +5,29 @@
         <h4 class="modal-title" id="password-modal-title">Password Required</h4>
       </div>
       <div class="modal-body">
-        This content requires a password to access. Your password is:
-        <pre id="hint">openvault-takes-security-seriously</pre>
-        Please enter it below.
+          <p>
+            To access this material, 
+            please read the terms and conditions,
+            indicate your agreement,
+            and provide the password. 
+          </p>
+          <p>
+            Your password is:
+            <script>document.write('<span id="hint">'+Math.random().toString(36).slice(2)+'</span>');</script>
+          </p>
+        <iframe src="/plain/terms-and-conditions" style="width: 100%;"></iframe>
       </div>
       <div class="modal-footer">
-        <input id="password" type="password"></input>
+        I have read and agree to the terms and conditions:
+        <input id="read-and-agree" type="checkbox"></input>
+        <input id="password" type="password" placeholder="Password from above"></input>
         <button type="submit" class="btn btn-primary" onclick="
-          if ($('#password').val() === $('#hint').text()) {
+          // When I copy-and-paste, I get a space character at the beginging...
+          // not sure why, but replace cleans it up.
+          if ($('#password').val().replace(/\W/g,'') === $('#hint').text() &amp;&amp; $('#read-and-agree').is(':checked')) {
```

Comment:
@mccalluc ah.. right. Curious, is it true that it might still work, just not validate as xhtml?